apps/api/src/app/environments/usecases/update-widget-settings/update-widget-settings.usecase.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common';
+import { EnvironmentRepository } from '@novu/dal';
+import { UpdateWidgetSettingsCommand } from './update-widget-settings.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { IWidgetSettings } from '@novu/dal';
+
+@Injectable()
+export class UpdateWidgetSettings {
+  constructor(private environmentRepository: EnvironmentRepository) {}
+
+  async execute(command: UpdateWidgetSettingsCommand) {
+    const environment = await this.environmentRepository.findById(command.environmentId);
+
+    if (!environment) {
+      throw new ApiException(`Environment id: ${command.environmentId} not found`);
+    }
+
+    const updateWidgetSettings: Partial<IWidgetSettings> = {};
+
+    if (command.widget.notificationCenterEncryption != null) {
+      updateWidgetSettings.notificationCenterEncryption = command.widget.notificationCenterEncryption;
```

Comment:
```suggestion
updateWidgetSettings['widget.notificationCenterEncryption'] = command.widget.notificationCenterEncryption
```

Reply:
Hey @scopsy,  is there a specific reason to use the square brackets property accessor the dot property accessor should work as good no?